### PR TITLE
Support deserializing from borrowed or owned bytes, too

### LINKED ISSUE #669: [primitive-types] support deserializing H256 et al from bytes as well as hex strings
H256 et al serialize to hex strings, presumably to make things a little nicer when working with JSON.

How about we add support to allow them to be deserialized from bytes as well as just hex strings, which is another natural way to represent these things in the serde data model, and allows for better interaction via crates like `scale-value` (and any other types that prefer to serialize to bytes rather than hex strings), eg: https://github.com/luketchang/scale-value-deser/blob/main/src/main.rs

### PR DESCRIPTION
**Pull request by jsdw**
Support deserializing from bytes, too (see #669).

Closes #669

### CODE REVIEW

File: primitive-types/impls/serde/src/serialize.rs
Comment (ordian) [inline primitive-types/impls/serde/src/serialize.rs:291]:
```suggestion
			bytes[..v.len()].copy_from_slice(v);
```

**Review by ordian — APPROVED** [review]
Could you also bump the patch version and update the changelog? Thanks!

**Review by niklasad1 — APPROVED** [review]

**Comment by luketchang** [comment]
Could we also add support for deserializing from a serde `sequence`? When substrate chain returns a storage query result, it comes in the form of a sequence of `U128` meant to be a `Vec<u8>`

**Comment by jsdw** [comment]
I've added support for sequences of bytes, too. That seemed natural and reasonable to support (and indeed it is what `scale_value::Value`s can work best with).

@ordian re version bumping and changelog:

The changelog has an unreleased seciton already and the version isn't bumped from that last change.

Should I either:
1. Bump the version, add my entry and turn "unreleased" into changelog for that version, or
2. Add my entry to "unreleased" and leave it to you or somebody else to bump the version and modify the changelog when they do a release?

**Comment by ordian** [comment]
> Should I either:
> 
>     1. Bump the version, add my entry and turn "unreleased" into changelog for that version, or
> 
>     2. Add my entry to "unreleased" and leave it to you or somebody else to bump the version and modify the changelog when they do a release?

I was suggesting 1, but if you're more comfortable with 2, that works too :)

**Comment by jsdw** [comment]
I'm happy with 1; happy to just publish the crate too once it's merged but wasnted to check that there wasn't any particular release approach/policy to follow if so :)

**Comment by ordian** [comment]
>  Perhaps I'll just publish this one first and can worry about updating the other crates in this repo that depend on it later; wdyt?

Yup, I can take care of the rest. It seems to be another semver-pocalypse which would necessitate a breaking bump for almost all crates.

**Comment by jsdw** [comment]
I was wondering whether we could have gotten away with a patch bump instead?

I've also been mulling over this and I think I'd like to add a followup PR to accomodate another deserialization woe (basically; the type info for types like H256 indicates that they are some newtype surrounding an array of bytes, and so it would be nice to try and support that, too, in the deserialization, rather than the current stuff which is all good but pretends that we're going straight to an array of bytes.)

**Comment by ordian** [comment]
> I was wondering whether we could have gotten away with a patch bump instead?

I've bumped patch versions on MSRV bump previously and people were not happy about that: #620. The MSRV is not caused by your PR, but an unreleased change.

I understand you want to get this out quickly and easily, but we gotta do it the right way.

> I've also been mulling over this and I think I'd like to add a followup PR to accomodate another deserialization woe (basically; the type info for types like H256 indicates that they are some newtype surrounding an array of bytes, and so it would be nice to try and support that, too, in the deserialization, rather than the current stuff which is all good but pretends that we're going straight to an array of bytes.)

It seems the crate hasn't been published, so feel free to submit additional PRs and change the date of the release in the changelog.

**Pull request merged**
